keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/ApplicationLoadBalancerHandler.kt

```diff
@@ -246,7 +246,10 @@ class ApplicationLoadBalancerHandler(
                       port = tg.port,
                       healthCheckEnabled = tg.healthCheckEnabled,
                       healthCheckTimeoutSeconds = Duration.ofSeconds(tg.healthCheckTimeoutSeconds.toLong()),
-                      healthCheckPort = tg.healthCheckPort,
+                      healthCheckPort = when (tg.healthCheckPort) {
+                        "traffic-port" -> tg.port
+                        else -> tg.healthCheckPort.toInt()
```

Comment:
can this might cause a class cast exception? (if it cannot be converted to int for some reason)

Reply:
Yeah, If clouddriver ever returned a string other than an integer or `"traffic-port"`, it would throw an exception that looks like this:

```
java.lang.NumberFormatException: For input string: "unexpected-string-here"
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:580)
	at java.lang.Integer.parseInt(Integer.java:615)
...
```

Since we never expect clouddriver to pass us a non-integer value, I think throwing an exception is the appropriate behavior here.